cpp/tests/join/conditional_join_tests.cu

```diff
@@ -167,7 +169,7 @@ struct ConditionalJoinPairReturnTest : public ConditionalJoinTest<T> {
     std::sort(result_pairs.begin(), result_pairs.end());
     std::sort(expected_outputs.begin(), expected_outputs.end());
 
-    EXPECT_TRUE(std::equal(result_pairs.begin(), result_pairs.end(), expected_outputs.begin()));
+    EXPECT_TRUE(std::equal(expected_outputs.begin(), expected_outputs.end(), result_pairs.begin()));
```

Comment:
I'm curious. Why was it prudent to swap the iteration around?

Reply:
The other order was hiding a bug where the join was returning an empty set. Since the first iterator terminated immediately, it never checked itemwise equality. The change isn't super important anymore since I'm now also verifying the size of the result, but it was made during an early iteration when I was trying to debug the problem.

Reply:
I have to remember this for my own tests.